### DISCUSSION TITLE
Import MongoDB query string and convert to querybuilder rules

### DISCUSSION THREAD

**Discussion by WebDevMichael** [react-querybuilder/react-querybuilder#338] [Ideas]
Hi,

React-Querybuilder is a very useful component for my hobby project, thank you for making this. 😄 

I'm working on a feature that will enable users importing a MongoDB query and convert this query string to querybuilder rules.

The basics are working, tests are being written. But I'm struggling a lot with Typescript types and also building issues.

**Comment by jakeboone02** [comment]
Hi, @WebDevMichael. We have an issue (#316) tracking the addition of some new `parse*` import methods, including `parseMongoDB`. Could you put any questions you have as comments on that issue? I've been meaning to get started on that one for a while but haven't actually done any work on it yet.

Internally, the TypeScript types for this library are a _little_ complex due to the support of `independentCombinators` which enforces a different structure on the query object.  If your intention is to contribute to `react-querybuilder`, I'd be happy to walk you through the types and answer any questions.

You should base the branch that you intend to make into a PR on the [`between-fields` branch](https://github.com/react-querybuilder/react-querybuilder/tree/between-fields) (see #337). It hasn't been merged into `main` yet but should be soon, and it has some relevant updates to the `parse*` methods.
